packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -417,8 +412,6 @@ export const usePickerValue = <
     setDate({
       value: valueManager.emptyValue,
       action: 'acceptAndClose',
-      // force `onChange` in cases like input (value) === `Invalid date`
-      forceOnChangeCall: !valueManager.areValuesEqual(utils, value as any, valueManager.emptyValue),
```

Comment:
🙌 The new solution for these cases seems way nicer! 👍 
Can we now ditch this prop/flag from the `setDate`? 🤔

Reply:
That was a nice side effect :ok_hand: 
I'll clean it :+1: